### ISSUE TITLE
Unity.call function does not survive multiple page loads on iOS and OS X

### ISSUE THREAD

**Issue by jonagill** [gree/unity-webview#107]
The Unity.call() function that is injected by SampleWebView.cs stops working on iOS and OS X if you change the page. (For instance, load to an initial page, click on a link, then hit the back button.)

This does not appear to be broken on Android, which seems to inject the same functionality through a different path.

This is fixed by manually opening the unity: in your JS, rather than using the Unity.call() function.

**Comment by KojiNakamaru** [comment]
Yes, the implementation of Unity.call() for OSX/iOS is fragile and not elegant, but sufficient in most cases in which a developer provides pages to be shown.

We could for example inject Unity.call() by repeatedly calling EvaluateJS() and/or utilize recent iOS/OSX WKWebView functionality to define methods exposed to JavaScript (similar one is provided by Android WebView and utilized). The former however consumes more CPUs and the latter cannot support older iOS/OSX. I'll try if I think up some better solution.

**Comment by sm-j** [comment]
I have got same issue.
I am adding login system with webview and on 1st page user fills their login details and then success page is loaded where i have added Unity.call in webpage.
This works fine in android version but when i try it in iOS device it doesn't get message passed in unity.call.

Any suggestions for the issue?

**Comment by KojiNakamaru** [comment]
cf. https://github.com/gree/unity-webview/issues/390#issuecomment-458006815
